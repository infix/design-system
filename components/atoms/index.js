import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import { Select } from './Select'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  Select,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
