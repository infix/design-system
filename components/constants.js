/* global module */
const defaultWeights = [
  400,
  700
]

const defaultBreakpoints = [
  40,
  52,
  64
]

const defaultSpace = [
  0,
  8,
  16,
  32,
  64
]

const defaultFontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72
]

module.exports = {
  defaultBreakpoints,
  defaultSpace,
  defaultFontSizes,
  defaultWeights
}
