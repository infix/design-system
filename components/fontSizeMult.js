/* global require */
import {
  style,
  px
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return style({
    prop: 'fontSize',
    cssProperty: propName,
    key: 'fontSizes',
    transformValue: n => px(n * k),
    scale: defaultFontSizes
  })
}

export default fontSizeMult
