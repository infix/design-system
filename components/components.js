import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */

export const Button = atoms.Button
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Select = atoms.Select

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Select,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
