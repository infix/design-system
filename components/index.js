import {
  Provider,
  Flex,
  Box
} from 'rebass'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Provider,
  Flex,
  Box
} from 'rebass'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  Link,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Label,
  Input,
  Select,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, { Provider, Box, Flex, colors, theme })

export default OONIComponents
