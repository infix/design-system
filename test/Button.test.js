/* global describe, test, expect */
import React from 'react'
import toJson from 'enzyme-to-json'
import { renderJSON, shallowWithTheme } from './index'
import { Button } from '../components'

describe('Button', () => {
  test('renders', () => {
    const wrapper = renderJSON(
      <Button />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('render with width prop', () => {
    const wrapper = shallowWithTheme(
      <Button width={1} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
    // XXX for some reason only a shallow render makes this assertion true
    expect(wrapper).toHaveStyleRule('width', '100%')
  })

  test('render with margin prop', () => {
    const wrapper = shallowWithTheme(
      <Button m={2} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
    expect(wrapper).toHaveStyleRule('margin', '8px')
  })

  test('render with fontSize prop', () => {
    const wrapper = shallowWithTheme(
      <Button fontSize={2} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
    expect(wrapper).toHaveStyleRule('font-size', '22px')
  })

  test('render with inverted prop', () => {
    const wrapper = shallowWithTheme(
      <Button inverted />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
    expect(wrapper).toHaveStyleRule('color', '#0588cb')
    expect(wrapper).toHaveStyleRule('background-color', '#f8f9fa')
  })

  test('render with hollow prop', () => {
    const wrapper = shallowWithTheme(
      <Button hollow />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
    //  expect(wrapper).toHaveStyleRule('background-color', 'transparent')
  })

  test('render with border prop (and hollow)', () => {
    const wrapper = shallowWithTheme(
      <Button hollow border={4} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
    expect(wrapper).toHaveStyleRule('border', '4px solid #0588cb')
  })

})
