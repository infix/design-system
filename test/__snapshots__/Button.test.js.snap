// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with border prop (and hollow) 1`] = `
.c0 {
  font-size: 16px;
}

<Button__StyledButton
  border={4}
  className="c0"
  fontSize={1}
  hollow={true}
  inverted={false}
/>
`;

exports[`Button render with fontSize prop 1`] = `
.c0 {
  font-size: 22px;
}

<Button__StyledButton
  border={2}
  className="c0"
  fontSize={2}
  hollow={false}
  inverted={false}
/>
`;

exports[`Button render with hollow prop 1`] = `
.c0 {
  font-size: 16px;
}

<Button__StyledButton
  border={2}
  className="c0"
  fontSize={1}
  hollow={true}
  inverted={false}
/>
`;

exports[`Button render with inverted prop 1`] = `
.c0 {
  font-size: 16px;
}

<Button__StyledButton
  border={2}
  className="c0"
  fontSize={1}
  hollow={false}
  inverted={true}
/>
`;

exports[`Button render with margin prop 1`] = `
.c0 {
  margin: 8px;
  font-size: 16px;
}

<Button__StyledButton
  border={2}
  className="c0"
  fontSize={1}
  hollow={false}
  inverted={false}
  m={2}
/>
`;

exports[`Button render with width prop 1`] = `
.c0 {
  width: 100%;
  font-size: 16px;
}

<Button__StyledButton
  border={2}
  className="c0"
  fontSize={1}
  hollow={false}
  inverted={false}
  width={1}
/>
`;

exports[`Button renders 1`] = `
.c1 {
  border-radius: 32px;
  display: inline-block;
  height: 40px;
  line-height: 1;
  padding-left: 32px;
  padding-right: 32px;
  text-transform: uppercase;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  outline: 0;
  font-family: inherit;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f8f9fa;
  background-color: #0588cb;
  border: none;
  text-align: center;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
  z-index: 1;
  -webkit-transition: .2s ease-out;
  transition: .2s ease-out;
  cursor: pointer;
}

.c1:hover {
  border: none;
  background-color: #37a6ed;
}

.c1:active {
  -webkit-transition: .2s ease-in;
  transition: .2s ease-in;
  border: none;
  background-color: #0f77b8;
}

.c0 {
  font-size: 16px;
}

<Button
  border={2}
  fontSize={1}
  hollow={false}
  inverted={false}
>
  <Button__StyledButton
    border={2}
    className="c0"
    fontSize={1}
    hollow={false}
    inverted={false}
  >
    <button
      className="c0 c1"
      fontSize={1}
    />
  </Button__StyledButton>
</Button>
`;
